homeassistant/components/openuv/config_flow.py

```diff
@@ -0,0 +1,65 @@
+"""Config flow to configure deCONZ component."""
+
+from collections import OrderedDict
+
+import voluptuous as vol
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.core import callback
+from homeassistant.const import (
+    CONF_API_KEY, CONF_ELEVATION, CONF_LATITUDE, CONF_LONGITUDE)
+from homeassistant.helpers import config_validation as cv
+
+from .const import DOMAIN
+
+
+@callback
+def configured_instances(hass):
+    """Return a set of configured OpenUV instances."""
+    return set(
+        '{0}, {1}'.format(
+            entry.data[CONF_LATITUDE], entry.data[CONF_LONGITUDE])
+        for entry in hass.config_entries.async_entries(DOMAIN))
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class OpenUvFlowHandler(data_entry_flow.FlowHandler):
+    """Handle an OpenUV config flow."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize the config flow."""
+        pass
+
+    async def async_step_import(self, import_config):
+        """Import a config entry from configuration.yaml."""
+        return await self.async_step_user(import_config)
+
+    async def async_step_user(self, user_input=None):
+        """Handle the start of the config flow."""
+        errors = {}
+
+        if user_input is not None:
+            identifier = '{0}, {1}'.format(
+                user_input.get(CONF_LATITUDE, self.hass.config.latitude),
+                user_input.get(CONF_LONGITUDE, self.hass.config.longitude))
+
+            if identifier not in configured_instances(self.hass):
```

Comment:
Please validate here too that the API key works.

Reply:
Both.

You should not create a config entry with an invalid API key. Instead notify the user that they entered the wrong API key.

During initialization, you still need to validate the API key as it might be a subsequent startup.

Reply:
Got it. I added a utility method to `pyopenuv` to check whether an API key is valid; I now use that both in `async_setup` and during the config entry flow (045bf63). Let me know if I'm mistaken anywhere.